### ISSUE TITLE
Survey tool popup

### ISSUE THREAD

**Issue by meh9** [TerriaJS/terriajs#2634]
There has been significant interest in knowing why and how the users use the data that a TerriaJS map makes available.

One way of collecting this data could be to have a simple "Survey" tool pop up and ask a user some very brief questions about a particular dataset they enabled. The tool would be very simple, have at most 1-3 questions, and take less than say 1 minute to complete.

We would have to work out some heuristics (probably customisable) for when to pop up the Survey tool such as:

* the user has a 10% chance of being selected at all
* the specific layer has been enabled for at least 3 minutes
* the user has been actively moving the view for 2 minute
* etc

Some example customisable questions that may be interesting for datasets, of which we may ask only one or two at the time to keep things fast and simple and more likely to get a response:

* You have been using layer X, how is that data useful in solving your problem today?
* Is there another dataset that would be useful to combine with data X?
* What is the problem you are trying to solve with the data you are currently viewing? <log the datasets>
* What functionality should we add to make using this data easier?
* etc

@kring @philipgrimmett @hilarycinis and others, comments?

**Comment by hilarycinis** [comment]
Thanks @meh9 
I support a survey, am also a fan of triangulating with qualitative research eg interviews or observed use to better understand context of use and technical ability. 

I think get rolling on the survey with @philipgrimmett .

Just to check - are these the actual things you are interested in learning? When you say "there has been significant interest" is that within the terria team or from sponsors?

**Comment by meh9** [comment]
@hilarycinis No, those are probably not the actual things we're interested in learning - we would have to figure out what the questions should actually be.

I fully agree that this should definitely not be the only way to find the answers, and it should definitely be paired with other ways of finding the answers such as interviews etc. The popup survey tool will definitely suffer from selection bias of people just wanting popups to go away without reading etc.
